code/data_structures/src/binary_heap/binary_heap.cpp

```
@@ -0,0 +1,175 @@
+/* Binary Heap in C++ */
+
+#include <iostream>
+#include <cstdlib>
+#include <vector>
+#include <iterator>
+
+
+class BinaryHeap{
+
+    private:
+        std::vector <int> heap;
+        int left(int parent);
+        int right(int parent);
+        int parent(int child);
+        void heapifyup(int index);
+        void heapifydown(int index);
+    public:
+        BinaryHeap()
+        {}
+        void Insert(int element);
+        void DeleteMin();
+        int ExtractMin();
+        void DisplayHeap();
+        int Size();
+};
+
+// Return Heap Size
+int BinaryHeap::Size(){
+    return heap.size();
+}
+ 
+// Insert Element into a Heap 
+void BinaryHeap::Insert(int element){
+    heap.push_back(element);
+    heapifyup(heap.size() -1);
+}
+
+// Delete Minimum Element 
+void BinaryHeap::DeleteMin(){
+    
+    if (heap.size() == 0){
+        std::cout << "Heap is Empty"<< std::endl;
```

Please raise an exception instead.